### PR TITLE
Fix `api.contact.in_contact`

### PR DESCRIPTION
**Pull request by flferretti**
The returned length of `links_in_contact` was always equal to the total number of links in the model, instead of being equal to the length of the input `link_names`

----
📚 Documentation preview 📚: https://jaxsim--119.org.readthedocs.build//119/

### CODE REVIEW

**Review by diegoferigo — CHANGES_REQUESTED** [review]
Thanks for having a look @flferretti. It is indeed wrongly returning always an array with $n_L$ elements. However, I fear your fix is not correct. I guess you need to create a boolean mask that selects the elements of `links_in_contact` corresponding to the passed `link_names`. Right now, you are selecting the first elements of `links_in_contact`, that would not correspond to `link_names`. Did I get it right?

File: src/jaxsim/api/contact.py [hunk 1]
```diff
@@ -190,11 +191,11 @@ def in_contact(
 
     links_in_contact = jax.vmap(
         lambda link_index: jnp.where(
-            model.kin_dyn_parameters.contact_parameters.body == link_index,
+            model.physics_model.gc.body == link_index,
             below_terrain,
             jnp.zeros_like(below_terrain, dtype=bool),
         ).any()
-    )(jnp.arange(model.number_of_links()))
+    )(Link.names_to_idxs(link_names=link_names, model=model))
```
Comment (diegoferigo) [inline src/jaxsim/api/contact.py:198]:
```suggestion
    )(js.link.names_to_idxs(link_names=link_names, model=model))
```

File: src/jaxsim/api/contact.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 import jaxsim.rbda
 import jaxsim.typing as jtp
 
+from . import link as Link
```
Comment (diegoferigo) [inline src/jaxsim/api/contact.py:10]:
```suggestion
```

File: src/jaxsim/api/contact.py [hunk 3]
```diff
@@ -190,11 +191,11 @@ def in_contact(
 
     links_in_contact = jax.vmap(
         lambda link_index: jnp.where(
-            model.kin_dyn_parameters.contact_parameters.body == link_index,
+            model.physics_model.gc.body == link_index,
```
Comment (diegoferigo) [inline src/jaxsim/api/contact.py:194]: The `functional` branch should no longer access `model.physics_model`.

```suggestion
            model.kin_dyn_parameters.contact_parameters.body == link_index,
```

**Review by diegoferigo — CHANGES_REQUESTED** [review]

File: src/jaxsim/api/contact.py (hunk 3 above)
Comment (diegoferigo) [reply src/jaxsim/api/contact.py:194]: Tests are not failind because this function is not yet part of any test.

Comment (flferretti) [reply src/jaxsim/api/contact.py]: Done in https://github.com/ami-iit/jaxsim/pull/119/commits/dd0d00b4c76315b48a2a096157ff6531e5e14037

File: src/jaxsim/api/contact.py (hunk 2 above)
Comment (flferretti) [reply src/jaxsim/api/contact.py]: Done in https://github.com/ami-iit/jaxsim/pull/119/commits/dd0d00b4c76315b48a2a096157ff6531e5e14037

File: src/jaxsim/api/contact.py (hunk 1 above)
Comment (flferretti) [reply src/jaxsim/api/contact.py]: Addressed in https://github.com/ami-iit/jaxsim/pull/119/commits/dd0d00b4c76315b48a2a096157ff6531e5e14037
